### PR TITLE
implementation of ACM Manager and EDM Manager as derived classes of c…

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview
 - EDM Manager basically moved to RestManager and EDMManager and ACMManager are derived classes. 
- UserMixin moved to RestManager.py, EndointMixin incorporated within RestManager (the Endpoint of a rest API is not a mixin, it's fundamental).
- Config of EDM tweaked and extended to support ACM. 
- Passthrough ACM written to do the same as for ECM 
- Annotations get used as a template to read from ACM

---
**REST API Updates**

Currently the annotations get returns all of the annotations on ACM but not the local ones, so a new interface was added to read just the local ones

```
[GET] /api/v1/annotations/local/
{
   
}
```

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [ x Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly covered
  - Example: The percentage of the code covered by tests has not decreased
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

File: app/extensions/restManager/RestManager.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+        # Must be overwritten by the derived class
+        assert self.NAME is not None
+        assert self.ENDPOINT_PREFIX is not None
+        assert self.ENDPOINTS is not None
+
+        self.app = app
+        self.targets = set([])
+
+        app.edm = self
```
Comment (hwindsor) [inline app/extensions/restManager/RestManager.py:95]: Well spotted, that's a bug :)

File: app/extensions/restManager/RestManager.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        ), f"Missing credentials for named {self.NAME} configs: {', '.join(missing_creds)}"
+
+        # Update the list of known named targets
+        self.targets.update(uris.keys())
+
+        # Assign local references to the configuration settings
+        self.uris = uris
+        self.auths = authns
+
+    def _init_sessions(self):
```
Comment (hwindsor) [inline app/extensions/restManager/RestManager.py:163]: Not certain yet if this needs to be in the base or derived class and if the 'session.login' needs to be configurable. Currently not working for ACM.

File: app/extensions/restManager/RestManager.py (hunk 1 above)
Comment (mmulich) [reply app/extensions/restManager/RestManager.py:95]: IMO this kind of attribute assignment to the app object should happen in the `init_app` space. I feel like it's a good idea for the app to only be used as reference and for registration during a manager object's init.

I'm also not a fan of `self.app = app` assignment. This kind of thing is frowned upon by in the Flask documentation, but lots of developers do it anyway. The better practice is to use the app as needed (or better yet a Blueprint), but not store it within the object. Instead use `flask.current_app` when you need to reference the current application in your object's logic.

Comment (hwindsor) [reply app/extensions/restManager/RestManager.py:95]: Those bits were just copied from the Existing EDMManager and tweaked but I see both your points. 
Fixed them. Will do a bit more testing today.

**Comment by hwindsor** [comment]
/

**Review by naknomum — APPROVED** [review]
seems solid.

**Pull request merged**